src/main/java/org/radarcns/phone/PhoneSensorManager.java

```diff
@@ -104,6 +137,30 @@ public void start(@NonNull final Set<String> acceptableIds) {
             logger.warn("Phone Light sensor not found");
         }
 
+        // Gyroscope
+        if (sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE) != null) {
+            Sensor gyroscope = sensorManager.getDefaultSensor(Sensor.TYPE_GYROSCOPE);
+            sensorManager.registerListener(this, gyroscope, SENSOR_DELAY_DEFAULT);
+        } else {
+            logger.warn("Phone Gyroscope not found");
+        }
+
+        // Magnetic field
+        if (sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD) != null) {
+            Sensor magneticField = sensorManager.getDefaultSensor(Sensor.TYPE_MAGNETIC_FIELD);
+            sensorManager.registerListener(this, magneticField, SENSOR_DELAY_DEFAULT);
+        } else {
+            logger.warn("Phone magnetometer not found");
+        }
+
+        // Steps
+        if (sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER) != null) {
+            Sensor stepCounter = sensorManager.getDefaultSensor(Sensor.TYPE_STEP_COUNTER);
```

Comment:
I think we should leave the unregister in. This is desired behaviour, as the participant should be able to 'stop' the tracking by closing the app completely (also the background service). As far as I can see, this is the only time the sensors are unregistered.

When the participant reopens the app, the tracking continues from that point onward.
Or are there other scenarios to consider?